hs_core/views/resource_metadata_rest_api.py

```diff
@@ -202,4 +204,5 @@ def put(self, request, pk):
 
         resource = hydroshare.get_resource_by_shortkey(shortkey=pk)
         serializer = resource.metadata.serializer
+        self.serializer_class = resource.metadata.serializer
```

Comment:
It may be a bit more readable to just do ```self.serializer_class = serializer```

Reply:
yes, that makes sense.  I'll update.